packages/twenty-front/src/modules/page-layout/components/PageLayoutGridLayout.tsx

```diff
@@ -97,10 +117,50 @@ export const PageLayoutGridLayout = ({ tabId }: PageLayoutGridLayoutProps) => {
   const isLayoutEmpty =
     !isDefined(activeTabWidgets) || activeTabWidgets.length === 0;
 
-  const layouts = isLayoutEmpty
+  const hasPendingPlaceholder =
+    isDefined(pageLayoutDraggedArea) && !isLayoutEmpty;
+
+  const baseLayouts = isLayoutEmpty
     ? EMPTY_LAYOUT
     : (pageLayoutCurrentLayouts[tabId] ?? EMPTY_LAYOUT);
 
+  const layouts = hasPendingPlaceholder
+    ? addPendingPlaceholderToLayouts(baseLayouts, pageLayoutDraggedArea)
+    : baseLayouts;
+
+  const widgets = [];
+
+  if (isLayoutEmpty) {
+    widgets.push(
```

Comment:
No that's a pattern to avoid, doing mutation at the root of a react component, in the react render loop is a bad idea, use a useMemo or an external hook but never this pattern.

Reply:
Moreover, let's avoid putting components into variable, that's a pattern that can ALWAYS be replaced by a simple .map on an array that will have been carefully crafted. 

Another reason is that we already keep all JSX-related code into the JSX return statement and that it is a convention that helps the reader avoid surprise JSX code into variables somewhere else in the code. 

JSX goes in the JSX part, array manipulation goes into the array manipulation logic, .map connects both, every time.